README.md

```diff
@@ -1,277 +1,1037 @@
+major trading pairs, this requirement for entry and exit liquidity is naturally incentivized to be
+satisfied.
+
+Since the current design is capable of handling 11-14 swaps in a single transaction, there is ample
+flexibility for finding arbitrage opportunities. For example, maybe the 3rd swap is at a slight loss
+but the 5th swap is such a good arbitrage that it more than makes up for it resulting in a net
+profit for the arbitrage.
+
+##### The Stablecoin <-> Stablecoin Two-Way Swap
+
+Currently, DApps treat stablecoins individually (ie, you may be able to borrow a loan in DJED but
+not USDC). This means, even though all stablecoins are practically the US dollar, the liquidity is
+fractured across all stablecoins. Two-way swaps change this.
+
+Imagine if Charlie has a swap for DJED <-> USDC where converting assets in either direction requires
+paying a 1% premium. The very existence of this swap, combined with the fact that Cardano-Swaps is
+freely composable with all DApps, means users can freely convert between stablecoins as needed. The
+liquidity would no longer be fractured. Alice could convert her DJED to USDC to buy an options
+contract that is asking for USDC. She just has to pay Charlie the 1% fee for the conversion.
+
+Since two-way swaps can go in either direction, these swaps do not need to be reset after each swap.
+For example, after Alice deposits DJED into the swap to claim USDC, Bob can now use the swap to
+convert his USDC to DJED by paying a 1% fee. Either way, Charlie makes 1% per swap and *is paid
+directly in the stablecoin being deposited*. And since the stablecoins are supposed to be 1:1 with
+the US dollar, the only risk is a peg-break. Anyone can be a liquidity provider and profit like
+Charlie does in this example.
+
+Being able to treat all stablecoins as a single stablecoin will be a game changer for DeFi.
+
+##### The Volatile Two-Way Swap
+
+While the stablecoin <-> stablecoin two-way swaps are the least risky, there is also an incentive to
+provide liquidity for volatile pairs. Consider two assets like ada and dust. Imagine if Alice thinks
+ada is going to appreciate against dust; therefore, she would rather have ada. Her prices can
+reflect this: converting ADA -> DUST could require a 15% premium while converting DUST -> ADA could
+require only a 5% premium. Would anyone pay the 15% premium?
+
+Yes, arbitragers would. Imagine if an arbitrager found a really good opportunity but they needed the
+ada. As long as the overall profit from the arbitrage can cover the 15% premium (and then some), the
+arbitrager may be more than willing to pay this high premium. Considering that the protocol supports
+chaining up to 14 swaps and illiquidity implies great arbitrage opportunities, it is actually likely
+that arbitragers will be able to find opportunities that justify paying this high premium. For
+example, let's say a single arbitrage yields a 5% profit and, since 14 swaps can fit in a single
```

Comment:
I think I should remove this abstract example. The following concrete example seems easier to understand and seems to make the case better. I don't think I need both examples.